lib/ansible/module_utils/basic.py

```diff
@@ -1695,6 +1695,14 @@ def atomic_move(self, src, dest, unsafe_writes=False):
                 # only try workarounds for errno 18 (cross device), 1 (not permitted),  13 (permission denied)
                 # and 26 (text file busy) which happens on vagrant synced folders and other 'exotic' non posix file systems
                 self.fail_json(msg='Could not replace file: %s to %s: %s' % (src, dest, to_native(e)), exception=traceback.format_exc())
+            elif e.errno == errno.EXDEV:
+                try:
+                    # copy() is the perfect equivalent of rename() when dealing with cross-device situations
+                    shutil.copy(b_src, b_dest)
+                except (IOError, OSError) as e:
+                    self.fail_json(msg='Could not replace file (using copy instead of rename): %s to %s: %s'
+                                       % (src, dest, to_native(e)),
+                                   exception=traceback.format_exc())
```

Comment:
this copy is both unnecessary and unwanted, it is not an atomic operation when it goes across partitions. 

Also the following code path does a 'copy and move' which does the same but in an atomic manner.